api/filtering/filtering.py

```diff
@@ -435,7 +435,13 @@ def query_filters(
     if provider_id:
         query_filters += ({"provider_id": {"eq": provider_id.casefold()}},)
     if updated_start or updated_end:
-        query_filters += _build_modified_on_filter(updated_start, updated_end)
+        # Need to check if date is less then 1970, since ElasticSearch doesn't
+        # allow timestamps before above this year. See ticket #ESSNTL-4458
+        if parser.isoparse(updated_start).year > 1970 or parser.isoparse(updated_end).year > 1970:
+            query_filters += _build_modified_on_filter(
+                updated_start=updated_start if parser.isoparse(updated_start).year > 1970 else None,
+                updated_end=updated_end if parser.isoparse(updated_end).year > 1970 else None,
```

Comment:
Do we need this redundant validation here?
```suggestion
                updated_start=updated_start
                updated_end=updated_end
```

Reply:
@jpramos123 yes, because we need to pass `None` to the parameter to avoid to create an filter with that, resulting with and empty: `modified_on: {}` on graphql query (in case both params are None).

Reply:
This logic could instead be put inside `_build_modified_on_filter`, which would be a bit cleaner (and the code would be simpler). The dates already get isoparsed at the beginning of that function, so you could just put the 1970 check in [this condition](https://github.com/RedHatInsights/insights-host-inventory/blob/master/api/filtering/filtering.py#L316), or between the `start > end` check and the part where the `modified_on_filter` gets populated

Reply:
I still think this suggestion would simplify the code and make the behavior more consistent ^